src/components/i18n/index.ts

```diff
@@ -0,0 +1,60 @@
+import defaultDictionary from './locales/en/messages.json';
+import * as _ from '../utils';
+import { I18nDictionary, Dictionary, DictValue } from '../../../types/configs/i18n-dictionary';
+
+/**
+ * This class will responsible for the translation through the language dictionary
+ */
+export default class I18n {
+  /**
+   * Property that stores messages dictionary
+   */
+  private static currentDictionary: I18nDictionary = defaultDictionary;
+
+  /**
+   * Perform translation of the string by namespace and a key
+   * If there is no translation found, returns passed key as a translated message
+   *
+   * @param namespace - path to translated string in dictionary
+   * @param dictKey - dictionary key. Better to use default locale original text
+   */
+  public static t(namespace: string, dictKey: string): string {
+    const section = I18n.getNamespace(namespace);
+
+    if (_.isEmpty(section)) {
+      _.log('I18n: section %o was not found in current dictionary', 'warn', namespace);
+    }
+
+    if (!section || !section[dictKey]) {
```

Comment:
Need to check here if `section[dictKey]` is an object (or not a string)

Reply:
Section is always an object

Reply:
As calls are not typesafe, you can pass path to value, not section

Reply:
so the section will be undefined, and this `if` cover this case